src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -46,14 +68,61 @@ export class WorkspaceIdConsumerWrapper {
           attributes,
           this.formatWorkspaceIdParams(wrapperOptions.request, options)
         ),
-      bulkCreate: <T = unknown>(
+      bulkCreate: async <T = unknown>(
```

Comment:
Seems these saved object type check was only exists in bulkCreate, shall we need add similar logic to create method?

Reply:
Good point, though user won't be able to visit data source management page in browser but they may curl the OSD API to create disallowed objects. Add that.